crud/common/rebalance.lua

```diff
@@ -0,0 +1,181 @@
+local fiber = require('fiber')
+local vshard_consts = require('vshard.consts')
+local utils = require('crud.common.utils')
+
+local MODULE_INTERNALS = '__module_crud_rebalance'
+local SETTINGS_SPACE_NAME = '_crud_settings'
+
+
+local M = rawget(_G, MODULE_INTERNALS)
+if not M then
+    M = {
+        safe_mode = false,
+        safe_mode_enable_hooks = {},
+        safe_mode_disable_hooks = {},
+        _router_cache_last_clear_ts = fiber.time()
+    }
+else
+    return M
+end
+
+local function create_space()
+    local settings_space = box.schema.space.create(SETTINGS_SPACE_NAME, {
+        engine = 'memtx',
+        format = {
+            { name = 'key', type = 'string' },
+            { name = 'value', type = 'any' },
+        },
+        if_not_exists = true,
+    })
+    settings_space:create_index('primary', { parts = { 'key' }, if_not_exists = true })
+end
+
+local function safe_mode_trigger(_, new, space, op)
+    if space ~= '_bucket' then
+        return
+    end
+    if (op == 'INSERT' and new.status == vshard_consts.BUCKET.RECEIVING) or
+            (op == 'REPLACE' and new.status == vshard_consts.BUCKET.SENDING) then
+        box.broadcast('_crud.safe_mode_enable', true)
```

Comment:
may be add constant for `_crud.safe_mode_enable` too?

Reply:
Added a constant